cpp/include/raft/spatial/knn/knn.cuh

```diff
@@ -151,17 +151,17 @@ inline void select_k(const value_t* in_keys,
       break;
 
     case SelectKAlgo::RADIX_8_BITS:
-      detail::topk::radix_topk<value_t, idx_t, 8, 512>(
+      matrix::detail::select::radix::select_k<value_t, idx_t, 8, 512>(
```

Comment:
Are we just using the `detail` version here because the function is deprecated? Why not use the public version?

Reply:
The implementation variants in radix- and warpsort- files weren't and were never meant to be public. In this PR, I've shuffled the definitions in the `detail` namespace to better organize them.

Reply:
Ah, I see. It's because we're accepting `SelectKAlgo` as an argument and we don't want to break this API by removing that. I'm okay with this, especially since this function is going away soon.